client/app/components/Chart/__tests__/ChartControl.spec.jsx

```diff
@@ -7,28 +7,79 @@ import React from 'react';
 import { ChartControl } from '../ChartControl';
 
 describe('ChartControl', () => {
+  // mocked data
+  const MOMENTS_TYPE = 'Moments';
+  const CATEGORIES_TYPE = 'Categories';
+  const MOODS_TYPE = 'Moods';
+  const types = [MOMENTS_TYPE, CATEGORIES_TYPE, MOODS_TYPE];
+  const data = {
+    [MOMENTS_TYPE]: {
+      '2013-02-10 00:00:00 -0800': 11,
+      '2013-02-11 00:00:00 -0800': 6,
+    },
+    [CATEGORIES_TYPE]: [
+      {
+        data: {
+          '2013-02-10': 3,
+          '2013-02-11': 6,
+          '2013-02-12': 13,
+        },
+        name: 'School',
+      },
+    ],
+    [MOODS_TYPE]: [
+      {
+        data: {
+          '2013-02-10': 2,
+          '2013-02-11': 8,
+          '2013-02-12': 3,
+        },
+        name: 'Anxious',
+      },
+    ],
+  };
+
   it('renders', () => {
     let wrapper = null;
     expect(() => {
       // shallow render only lists the first level of components
       wrapper = shallow(
         <ChartControl
-          types={['Moments']}
-          initialParams={{
-            type: 'Moments',
-            data: {
-              Moments: {
-                '2013-02-10 00:00:00 -0800': 11,
-                '2013-02-11 00:00:00 -0800': 6,
-              },
-            },
-          }}
+          types={types}
+          initialParams={{ type: MOMENTS_TYPE, data }}
         />,
       );
     }).not.toThrow();
 
     expect(wrapper).not.toBeNull();
-    expect(wrapper.find('ChartControlButton').length).toEqual(1);
+    expect(wrapper.find('ChartControlButton').length).toEqual(types.length);
     expect(wrapper.find('Chart').length).toEqual(1);
   });
+
+  it('passes the expected data for the selected type', () => {
+    const wrapper = shallow(
+      <ChartControl
+        types={types}
+        initialParams={{ type: MOMENTS_TYPE, data }}
+      />,
+    );
+
+    // query child chart component to get current props
+    let chart = wrapper.find('Chart');
+    // test initial data passed down to the chart
+    expect(chart.prop('data')).toBe(data[MOMENTS_TYPE]);
+
+    // simulate click on ChartControlButton for a specified type
+    const typeToTest = types[Math.floor(Math.random() * types.length)];
```

Comment:
Why not just specify a type instead of randomizing it?

Reply:
I...lack a concrete rationale for it :smile: I'd even specified `'Categories'` initially, but then got wondering if the test might be more generic if it didn't matter which type was selected?

Would it be better to specify a fixed value? I can include that in my updates as well 👍 

PS: I made a similar change for the expected number of chart control buttons too, tying it to `types.length`. I can also change that back to a constant number too if it makes more sense to do so.

Reply:
I think it's fine to specify a fixed value, that way the assertion is more "reliable".

Reply:
Sounds good! I'll change these back

Reply:
On a related note, would it be better to keep the mock data specific to each test too, as opposed to the current update? I can revert the first test back to its original state and follow a similar pattern for the second test?